File: le-spark/src/test/resources/scripts/join.py
# script_input = {}
# script_output = {}

import pyspark.sql.functions as sf

table1 = lattice.input[0]
table2 = lattice.input[1]

join_key = lattice.params["JOIN_KEY"]
agg_key = lattice.params["AGG_KEY"]

df = table1.join(table2, table1[join_key] == table2[join_key], "outer").groupBy(table2[join_key])
out1 = df.count().withColumnRenamed("count", "Cnt")
out2 = df.agg(sf.max(table1[agg_key]).alias("Max1"), sf.max(table2[agg_key]).alias("Max2"))

lattice.output = [out1, out2]
lattice.output_str = "This is my output!"

File: le-spark/src/test/resources/scripts/join.scala
//import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
//import com.fasterxml.jackson.module.scala.DefaultScalaModule
//import com.fasterxml.jackson.module.scala.experimental.ScalaObjectMapper
//import org.apache.spark.sql.DataFrame
//import org.apache.spark.sql.functions._
//val mapper = new ObjectMapper() with ScalaObjectMapper
//mapper.registerModule(DefaultScalaModule)
//class LatticeContext(val input: List[DataFrame], val params: JsonNode, val targets: List[JsonNode]) {
//  var output: List[DataFrame] = List[DataFrame]()
//}
//val lattice: LatticeContext = new LatticeContext(Nil, null, List[JsonNode]())

// ============
// BEGIN SCRIPT
// ============

// read input
val table1: DataFrame = lattice.input(0)
val table2: DataFrame = lattice.input(1)

// read config
val joinKey = lattice.params.get("JOIN_KEY").asText()
val aggKey = lattice.params.get("AGG_KEY").asText()

// computation
val df = table1.join(table2, table1(joinKey) === table2(joinKey), "outer").groupBy(table2(joinKey))
val out1 = df.count().withColumnRenamed("count", "Cnt")
val out2 = df.agg(max(table1(aggKey)).as("Max1"), max(table2(aggKey)).as("Max2"))

// write output
lattice.output = out1::out2::Nil
lattice.outputStr = "This is my output!"
